Make DOM Range correctly account for CDATASection nodes

annevk opened the pull request:
#### d1a819848ad3af8938fe1e25d65518383e551dae
<pre>
Make DOM Range correctly account for CDATASection nodes
<a href="https://bugs.webkit.org/show_bug.cgi?id=268903">https://bugs.webkit.org/show_bug.cgi?id=268903</a>

Reviewed by Ryosuke Niwa.

When the DOM Standard talks about Text nodes it is inclusive of
CDATASection nodes, e.g.,
<a href="https://dom.spec.whatwg.org/#concept-range-insert">https://dom.spec.whatwg.org/#concept-range-insert</a>

Test changes are exported via:
<a href="https://github.com/web-platform-tests/wpt/pull/44432">https://github.com/web-platform-tests/wpt/pull/44432</a>

* LayoutTests/imported/w3c/web-platform-tests/dom/common.js:
(setupRangeTests):
(nodeLength):
(myExtractContents):
(myInsertNode):
(isText):
(ensurePreInsertionValidity):
* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Node-compareDocumentPosition-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Node-contains-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Node-properties-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Node-properties.html:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-cloneContents-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-cloneContents.html:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-cloneRange-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-collapse-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-commonAncestorContainer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-compareBoundaryPoints-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-comparePoint-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-deleteContents-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-deleteContents.html:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-extractContents-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-insertNode-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-intersectsNode-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-isPointInRange-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-selectNode-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-set-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-surroundContents-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-surroundContents.html:
* LayoutTests/imported/w3c/web-platform-tests/dom/traversal/NodeIterator-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/traversal/NodeIterator-removal-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/traversal/TreeWalker-expected.txt:
* Source/WebCore/dom/Range.cpp:
(WebCore::Range::insertNode):
(WebCore::Range::surroundContents):

Canonical link: <a href="https://commits.webkit.org/274346@main">https://commits.webkit.org/274346@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7edf7499f1196b54a73f91961faec6f09b755c0b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38634 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17564 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/40985 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/41165 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/34293 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/40939 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/20368 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/14911 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/32470 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39207 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14804 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/33602 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12863 "Build is being retried. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; 1 api test failed or timed out; api tests running") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12844 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/34473 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42440 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35097 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34878 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38686 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/13450 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11153 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36890 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15062 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/33766 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8690 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13908 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/14538 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on LayoutTests/imported/w3c/web-platform-tests/dom/idlharness-shadowrealm.window-expected.txt:
```diff
@@ -1,134 +1,4 @@
-PASS XPathNSResolver interface: existence and properties of interface object
-PASS XPathEvaluator interface: existence and properties of interface object
-PASS XSLTProcessor interface: existence and properties of interface object
+Harness Error (FAIL), message = ReferenceError: Can't find variable: ShadowRealm
```
What's happening here? Seems we lost a LOT of test coverage so not great.

annevk replied:
Oh my bad, I'll exclude this file. It's annoyingly generated for dom/ and url/ and some other directories when you do --reset-results. Given that EWS isn't complaining I kinda suspect it just needs to be cleaned up, but not as part of this PR.

annevk replied:
It's exactly as I feared:
```
LayoutTests/TestExpectations:imported/w3c/web-platform-tests/compression/idlharness-shadowrealm.window.html [ Pass Failure ]
LayoutTests/TestExpectations:imported/w3c/web-platform-tests/dom/idlharness-shadowrealm.window.html [ Pass Failure ]
LayoutTests/TestExpectations:imported/w3c/web-platform-tests/encoding/idlharness-shadowrealm.window.html [ Pass Failure ]
LayoutTests/TestExpectations:imported/w3c/web-platform-tests/hr-time/idlharness-shadowrealm.window.html [ Pass Failure ]
LayoutTests/TestExpectations:imported/w3c/web-platform-tests/html/dom/idlharness-shadowrealm.window.html [ Pass Failure ]
LayoutTests/TestExpectations:imported/w3c/web-platform-tests/performance-timeline/idlharness-shadowrealm.window.html [ Pass Failure ]
LayoutTests/TestExpectations:imported/w3c/web-platform-tests/streams/idlharness-shadowrealm.window.html [ Pass Failure ]
LayoutTests/TestExpectations:imported/w3c/web-platform-tests/url/idlharness-shadowrealm.window.html [ Pass Failure ]
LayoutTests/TestExpectations:imported/w3c/web-platform-tests/user-timing/idlharness-shadowrealm.window.html [ Pass Failure ]
```
Maybe I'll take some time to clean this up as I keep running into this nonsense.

annevk committed:
Make DOM Range correctly account for CDATASection nodes
https://bugs.webkit.org/show_bug.cgi?id=268903

Reviewed by Ryosuke Niwa.

When the DOM Standard talks about Text nodes it is inclusive of
CDATASection nodes, e.g.,
https://dom.spec.whatwg.org/#concept-range-insert

Test changes are exported via:
https://github.com/web-platform-tests/wpt/pull/44432

* LayoutTests/imported/w3c/web-platform-tests/dom/common.js:
(setupRangeTests):
(nodeLength):
(myExtractContents):
(myInsertNode):
(isText):
(ensurePreInsertionValidity):
* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Node-compareDocumentPosition-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Node-contains-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Node-properties-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/nodes/Node-properties.html:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-cloneContents-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-cloneContents.html:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-cloneRange-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-collapse-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-commonAncestorContainer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-compareBoundaryPoints-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-comparePoint-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-deleteContents-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-deleteContents.html:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-extractContents-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-insertNode-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-intersectsNode-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-isPointInRange-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-selectNode-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-set-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-surroundContents-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/ranges/Range-surroundContents.html:
* LayoutTests/imported/w3c/web-platform-tests/dom/traversal/NodeIterator-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/traversal/NodeIterator-removal-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/traversal/TreeWalker-expected.txt:
* Source/WebCore/dom/Range.cpp:
(WebCore::Range::insertNode):
(WebCore::Range::surroundContents):

Canonical link: https://commits.webkit.org/274346@main
